### PR TITLE
feat: basket transactions sign

### LINKED ISSUE #1813: Basket. Sign operations screen
* Sign single operation (usual flow)
* Sign several operations
* Show modal with amount of completed transactions
* Save chain errors for uncompleted transactions
* Remove completed transactions

### LINKED ISSUE #1835: Basket. Update list of operations screen
1. Update shimmering when validating
2. Update displaying validation error 
3. Checkbox should be disabled for invalid opreation.

### LINKED ISSUE #1892: Basket. Change Validators. There is no validators in the list on confirmation page
## Description
When user create change validators staking operation, confirmation screen via basket has no info about validators
Screenshot 2024-06-25 at 13 26 35

## Steps to Reproduce

1. As wallet_pv add to basket change validators operation
2. Go to basket
3. Select operation to sign
4. Check validators on confirm page

## Expected Behavior
Validators data is available

## Actual Behavior
Empty list of validators

## Environment
https://github.com/novasamatech/nova-spektr/commit/f7cec56f1465ce4b53bff096241e125e0c8e9c14

### LINKED ISSUE #1896: Basket. Can't sign Unstake operation
## Description
When user add Unstake operation to the basket - he can't sign it. Operation with batch can't be opened, simple operation with unbound amount - can't pass verifications.

Example account: 5Cad3FdyS4j4z39uzTPhcPFey5MNXEszCCTTfkH6Ly1LsFwz

Screenshot 2024-06-26 at 10 22 29

Screenshot 2024-06-26 at 12 50 56

## Steps to Reproduce

1. Create Unstake operation with chill (keep bond amount less than minimal nominator deposit)
2. Add this operation to basket
3. Create Unstake operation without chill
4. Add this operation to basket
5. Go to basket
6. Try to sign

## Expected Behavior
Can sign this operations

## Actual Behavior
- Can't open operation with chill
- Simple operation is shown as Unavailable

## Environment
https://github.com/novasamatech/nova-spektr/commit/241bff1698bb2f040fdd1dddda5e557fa7040f72

<details>
  <summary>Example</summary>

https://github.com/novasamatech/nova-spektr/assets/40560660/09852423-7846-43be-ab6c-295c0800b67a

</details>

## Test data
[accounts.json](https://github.com/user-attachments/files/15985976/accounts.json)
[basketTransactions.json](https://github.com/user-attachments/files/15985977/basketTransactions.json)
[wallets.json](https://github.com/user-attachments/files/15985978/wallets.json)

### LINKED ISSUE #1899: Basket. Can't remove shimmering operations.
## Description
When user has shimmering operations in his basket - he can't remove it

<details>
  <summary>Example</summary>

https://github.com/novasamatech/nova-spektr/assets/40560660/8d2efbb4-03ca-4eda-9d9e-1fe557961e2a

</details>

## Steps to Reproduce

1. Add operations to the basket
2. Make some connection trouble
3. Try to delete operation with shimmering

## Expected Behavior
Can delete operation even if it's shimmering (loading state)

## Actual Behavior
Can't delete that type of operations

## Environment
https://github.com/novasamatech/nova-spektr/commit/399351495f8cedd4b5a0f0de5b8e6c243f521f57

### PR DESCRIPTION
**Pull request by Asmadek**
* update confirm screens to support several transactions
* sign several basket transactions
* fix sign several staking transactions

### CODE REVIEW

File: src/renderer/features/operations/OperationsConfirm/Withdraw/ui/Confirmation.tsx [hunk 1]
```diff
@@ -13,18 +13,24 @@ import { AccountsModal } from '@entities/staking';
...
-  const confirmStore = useUnit(confirmModel.$confirmStore);
-  const initiatorWallet = useUnit(confirmModel.$initiatorWallet);
-  const signerWallet = useUnit(confirmModel.$signerWallet);
-  const proxiedWallet = useUnit(confirmModel.$proxiedWallet);
+  const stores = useUnit(confirmModel.$confirmStore);
+  const initiatorWallets = useUnit(confirmModel.$initiatorWallets);
+  const signerWallets = useUnit(confirmModel.$signerWallets);
+  const proxiedWallets = useUnit(confirmModel.$proxiedWallets);
+
+  const confirmStore = stores?.[id];
```
Comment (johnthecat) [inline src/renderer/features/operations/OperationsConfirm/Withdraw/ui/Confirmation.tsx]: Why not [`useStoreMap`](https://effector.dev/en/api/effector-react/usestoremap/)? Should be a lot simpler.
```ts
const confirmStore = useStoreMap({
  store: confirmModel.$confirmStore,
  keys: [id], 
  fn: (x, [id]) => x?.[id]
})
```

**Review by pgolovkin — APPROVED** [review]

File: src/renderer/entities/basket/model/basket-model.ts [hunk 2]
```diff
@@ -17,6 +18,10 @@ const addBasketTxsFx = createEffect(
   },
 );
 
+const updateBasketTxsFx = createEffect(async (transactions: BasketTransaction[]): Promise<number[] | undefined> => {
```
Comment (tuul-wq) [inline src/renderer/entities/basket/model/basket-model.ts]:
```suggestion
const updateBasketTxsFx = createEffect((transactions: BasketTransaction[]): Promise<number[] | undefined> => {
```

File: src/renderer/entities/transaction/ui/Scanning/ScanMultiframeQr.tsx [hunk 3]
```diff
@@ -7,21 +7,18 @@ import { useI18n } from '@app/providers';
...
 
 type Props = {
-  api: ApiPromise;
-  chainId: ChainId;
-  accounts: Account[];
-  addressPrefix: number;
-  rootAddress?: string;
-  transactions: Transaction[];
+  apis: Record<ChainId, ApiPromise>;
+  signingPayloads: SigningPayload[];
```
Comment (tuul-wq) [inline src/renderer/entities/transaction/ui/Scanning/ScanMultiframeQr.tsx:21]: 👍

File: src/renderer/features/operations/OperationSign/model/sign-model.ts [hunk 4]
```diff
@@ -35,15 +23,26 @@ const formSubmitted = createEvent<SignatureData>();
...
+const $apis = combine(
   {
     apis: networkModel.$apis,
     store: $signStore,
   },
   ({ apis, store }) => {
-    return store?.chain ? apis[store.chain.chainId] : undefined;
+    if (!store) return {};
+
+    return store.signingPayloads.reduce((acc, payload) => {
```
Comment (tuul-wq) [inline src/renderer/features/operations/OperationSign/model/sign-model.ts]:
```suggestion
    return store.signingPayloads.reduce<Record<ChainId, ApiPromise>>((acc, payload) => {
```

File: src/renderer/features/operations/OperationsConfirm/AddProxy/model/confirm-model.ts [hunk 5]
```diff
@@ -19,67 +21,118 @@ type Input = {
...
-  { skipVoid: false },
 );
 
 export const confirmModel = {
-  $confirmStore,
-  $initiatorWallet,
-  $signerWallet,
-  $proxiedWallet,
-  $api,
+  $confirmStore: $confirmStore.map(
```
Comment (tuul-wq) [inline src/renderer/features/operations/OperationsConfirm/AddProxy/model/confirm-model.ts]: This is barely readable, let's raise readability of this property by extracting it into combine that can occupy more space

File: src/renderer/features/operations/OperationsConfirm/AddProxy/ui/Confirmation.tsx [hunk 6]
```diff
@@ -12,20 +12,44 @@ import { AssetBalance } from '@entities/asset';
...
+    fn: (value, [id]) => value?.[id],
+  });
+
+  const signerWallet = useStoreMap({
+    store: confirmModel.$signerWallets,
+    keys: [id],
+    fn: (value, [id]) => value?.[id],
+  });
+
+  const proxiedWallet = useStoreMap({
```
Comment (tuul-wq) [inline src/renderer/features/operations/OperationsConfirm/AddProxy/ui/Confirmation.tsx:42]: 👍

File: src/renderer/shared/lib/utils/substrate.ts [hunk 7]
```diff
@@ -44,7 +45,8 @@ export const createTxMetadata = async (
     blockNumber: block.header.number.toNumber(),
     genesisHash: genesisHash.toString(),
     metadataRpc: metadataRpc.toHex(),
-    nonce: nonce.toNumber(),
+    nonce: nonce.toNumber() + nonceDelta,
```
Comment (tuul-wq) [inline src/renderer/shared/lib/utils/substrate.ts:49]: Why `nonceDelta` was required to be added? Just curious 😌

File: src/renderer/shared/ui/Modals/BaseModal/BaseModal.tsx [hunk 8]
```diff
@@ -40,9 +42,10 @@ export const BaseModal = ({
         <div className="fixed inset-0 overflow-hidden flex min-h-full items-center justify-center p-4">
           <ModalTransition>
             <Dialog.Panel
+              style={panelStyle}
               className={cnTw(
                 'transform rounded-lg bg-white text-left align-middle shadow-modal transition-all w-[440px]',
-                panelClass,
+                panelClass ?? panelClass,
```
Comment (tuul-wq) [inline src/renderer/shared/ui/Modals/BaseModal/BaseModal.tsx]: `panelClass` or `panelClass`? Am I missing something?

**Review by tuul-wq — COMMENTED** [review]
Nice PR 👌

File: src/renderer/shared/lib/utils/substrate.ts (hunk 7 above)
Comment (Asmadek) [reply src/renderer/shared/lib/utils/substrate.ts:49]: If we execute several txs from one account, we have to upgrade nonce for all accounts, added it to function arguments comments

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**